design/custom-ca-support.md

```diff
@@ -0,0 +1,107 @@
+# Custom CA Bundle Support for S3 Object Storage
+
+It is desired that Velero performs SSL verification on the Object Storage
+endpoint (BackupStorageLocation), but it is not guaranteed that the Velero
+container has the endpoints' CA bundle in it's system store. Velero needs to
+support the ability for a user to specify custom CA bundles at installation
+time and Velero needs to support a mechanism in the BackupStorageLocation
+Custom Resource to allow a user to specify a custom CA bundle. This mechanism
+needs to also allow Restic to access and use this custom CA bundle.
+
+## Goals
```

Comment:
Does this support adding the certs on the client side? I know velero client-side commands such as `velero (backup|restore) (describe|logs)` on AWS/GCP/Azure get a 1-time use URL for retrieving files from object store, and on Minio they use a BSL's `Spec.Config.publicUrl` value in order to get non-encrypted values.

Reply:
It currently does not but I wouldn't rule that out. Are you thinking that the `velero install` command would install the cert on the client's machine? Or that the velero client commands would have a new flag `--cacert` which a user can optionally specify?

Reply:
I'm not exactly sure, honestly. I think either would work, so long as it's documented. A `--cacert` flag would probably be easier since certs can be managed in several different ways depending on client OS.

Mostly, I'd like to avoid users hitting issues like this one often. https://github.com/vmware-tanzu/velero/issues/2256